src/common/database.c

```diff
@@ -967,6 +967,29 @@ static int _upgrade_library_schema_step(dt_database_t *db, int version)
 
     sqlite3_exec(db->handle, "COMMIT", NULL, NULL, NULL);
     new_version = 15;
+  }
+  else if(version == 15)
+  {
+    sqlite3_exec(db->handle, "BEGIN TRANSACTION", NULL, NULL, NULL);
+    ////////////////////////////// custom image order
+    TRY_EXEC("ALTER TABLE main.images ADD COLUMN position INTEGER",
+             "[init] can't add `position' column to images table in database\n");
+    TRY_EXEC("CREATE INDEX main.image_position_index ON images (position)",
+             "[init] can't create index for custom image order table\n");
+
+    // Set the initial image sequence. The image id - the sequece images were imported -
+    // defines the initial order of images.
+    //
+    // An int64 is used for the position index. The upper 31 bits define the initial order.
```

Comment:
That's 32, right?

Reply:
I think not. [sign bit][31bit initial order][32 bit custom order] as only int64 can be used